### DISCUSSION TITLE
FileRedirectionFixup ProgramData

### DISCUSSION THREAD

**Discussion by dan-hughes** [TimMangan/MSIX-PackageSupportFramework#17] [Q&A] [answered]
Hi,

I have an old what looks like Winforms app that I do not have the source for. I'm trying to get it packaged up to use with AVD and get an error that the program is trying to read from ProgramData.

Upon my travels I ended up here and found this much more developed version of PSF. 

The application has some directories under C:\ProgramData\MyCompany, these exist in the MSIX package under VFS\Common AppData. I just need the PSF to redirect it for me.

Below is my config.json which doesn't seem to be working. What am I doing wrong?

``` JSON
{
    "applications": [
        {
            "id": "FCLIENT",
            "executable": "VFS/ProgramFilesX86/MyCompany/FClient/FClient.exe",
            "workingDirectory": "VFS/ProgramFilesX86/MyCompany/FClient/"
        }
    ],
    "processes": [
        {
            "executable": "^PsfLauncher.*",
            "fixups": [
                {
                    "dll": "FileRedirectionFixup32.dll",
                    "config": {
                        "redirectedPaths": {
                            "knownFolders": [
                                {
                                    "id": "ProgramData",
                                    "relativePaths": [
                                        {
                                            "base": "MyCompany",
                                            "patterns": [
                                                ".\\"
                                            ]
                                        }
                                    ]
                                }
                            ]
                        }
                    }
                }
            ]
        }
    ]
}
```

**Comment by dan-hughes** [comment, marked as answer]
Typically, I answered my own question moments after asking someone. For anyone else it needs to target the executable as defined [here](https://github.com/TimMangan/MSIX-PackageSupportFramework/wiki/Fixup:-FileRedirectionFixup).

**Reply by TimMangan** [reply]
Thanks for finding it.  That's where I was going to point you.  

By the way, PsfTooling (free app in Microsoft Store or on my TMurgent.com website) can be used while running the MSIX Packaging Tool to detect the need and apply the needed PSF components and configuration for you.  There is also a more capable paid for version on my website for post package detection/fixing of the packages.
